recipes/hdf4/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+        if self.options.jpegturbo:
+            self.requires.add("libjpeg-turbo/2.0.4")
+        else:
+            self.requires.add("libjpeg/9d")
+        if self.options.szip_support == "with_libaec":
+            self.requires.add("libaec/1.0.4")
+        elif self.options.szip_support == "with_szip":
+            self.requires.add("szip/2.1.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("hdf-" + self.version, self._source_subfolder)
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["HDF4_EXTERNALLY_CONFIGURED"] = True
+        self._cmake.definitions["HDF4_EXTERNAL_LIB_PREFIX"] = ""
+        self._cmake.definitions["HDF4_NO_PACKAGES"] = True
+        self._cmake.definitions["ONLY_SHARED_LIBS"] = self.options.shared
+        self._cmake.definitions["HDF4_ENABLE_COVERAGE"] = False
+        self._cmake.definitions["HDF4_ENABLE_DEPRECATED_SYMBOLS"] = True
+        self._cmake.definitions["HDF4_ENABLE_JPEG_LIB_SUPPORT"] = True # HDF can't compile without libjpeg or libjpeg-turbo
+        self._cmake.definitions["HDF4_ENABLE_Z_LIB_SUPPORT"] = True # HDF can't compile without zlib
+        self._cmake.definitions["HDF4_ENABLE_SZIP_SUPPORT"] = bool(self.options.szip_support)
+        self._cmake.definitions["HDF4_ENABLE_SZIP_ENCODING"] = self.options.get_safe("szip_encoding") or False
+        self._cmake.definitions["HDF4_PACKAGE_EXTLIBS"] = False
+        self._cmake.definitions["HDF4_BUILD_XDR_LIB"] = True
+        self._cmake.definitions["BUILD_TESTING"] = False
+        self._cmake.definitions["HDF4_INSTALL_INCLUDE_DIR"] = os.path.join(self.package_folder, "include", "hdf4")
+        self._cmake.definitions["HDF4_BUILD_FORTRAN"] = False
+        self._cmake.definitions["HDF4_BUILD_UTILS"] = False
+        self._cmake.definitions["HDF4_BUILD_TOOLS"] = False
+        self._cmake.definitions["HDF4_BUILD_EXAMPLES"] = False
+        self._cmake.definitions["HDF4_BUILD_JAVA"] = False
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        os.remove(os.path.join(self.package_folder, "lib", "libhdf4.settings"))
+
+    def package_info(self):
+        self.cpp_info.libs = self._get_ordered_libs()
+        self.cpp_info.includedirs.append(os.path.join(self.package_folder, "include", "hdf4"))
+        if self.options.shared:
+            self.cpp_info.defines.append("H4_BUILT_AS_DYNAMIC_LIB")
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs = ["m"]
```

Comment:
I'm starting to think that this should be changed to:
```suggestion
        if not self.options.shared:
            if self.settings.os == "Linux":
                self.cpp_info.system_libs = ["m"]
```
Because the shared binaries are already linked to the system libraries.
When linking to some shared libraries of your linux distribution of choice, you also do not have to link to some extra system libraries.
@uilianries @danimtb @SSE4  What do you think?

Reply:
I was thinking the same thing...

Reply:
Fair enough. Since Linux uses ldconfig for listing the runtime libraries.

Reply:
@uilianries I have a WIP `post_package` hook at https://github.com/madebr/hooks/tree/missing_systemlibs that checks missing system_libs.
For gcc-like toolchains, it uses `objdump`  (see `man ldd` for the why)
It is currently incomplete because of https://github.com/conan-io/conan/issues/6649

Reply:
this might not be true in all cases. for C library I believe it's (almost) always fine. for C++ project, especially which use heavy template code, some template implementations may contain calls to system libraries (e.g. libm), thus it's not sufficient to just link with shared library in general case.

Reply:
Indeed, it depends on the library interface (visibility in the includes).
But it should be up to the recipe author to check that it's ok.

Reply:
@SSE4 Well, observed. let's leave it as it is.